Add the `js_accordion` template using Handorgel.js

aschempp opened the pull request:
As discussed in https://github.com/contao/contao/pull/2301 quite a while ago, this implements an accordion using Handorgel.js (see https://github.com/contao/contao/pull/2301#issuecomment-906402752).

I tested this in the official Contao demo by simply selecting the `js_accordion` template and disabling the `j_accordion` template.

Before this can be merged we might discuss where to place the handorgel script (not sure if we want to add yet another `contao-components/handorgel`) 🙃

aschempp committed:
Added js_accordion template using Handorgel.js

leofeyer commented:
> not sure if we want to add yet another `contao-components/handorgel` 🙃

Yes, that is what we would have to do. 😄

But are we sure that we want to use Handorgel.js? It seems that there has never been a stable version (with a version number > 1) and the current version `0.5.0` has been released 3 years ago.

aschempp committed:
Remove unnecessary variable

aschempp commented:
> But are we sure that we want to use Handorgel.js? It seems that there has never been a stable version (with a version number > 1) and the current version 0.5.0 has been released 3 years ago.

I noticed that too recently. But I don't know of any similar script that works just as well and is as accessible. There's an open PR about one issue. I wonder if we should have a _Contao accordion script_ instead of a _fork of handorgel.js_

aschempp commented:
As discussed on the Contao Call today, we must make sure the content is visible if no javascript is available. The content is currently hidden due to the default CSS setting `display:none`.

leofeyer commented:
As discussed in the call, Handorgel.js does not show the contents when JavaScript is disabled. If we can solve this via CSS, the pull request could be merged. Otherwise we would have to discuss again or look for another script.

aschempp committed:
Simplify CSS and fix visibility if JavaScript is disabled

aschempp commented:
I have fixed the discussed visibility issue by simply adding an additional `handorgel` CSS class to the container element. This allows to target the content if the JS is loaded, which makes sense anyway.

I think we create the discussed `contao-components/handorgel` repository. It should contain the latest distribution javascript file and a CSS file containing the lines I added to the template.

leofeyer commented:
The component is ready: https://github.com/contao-components/handorgel

aschempp committed:
Use the component

ausi commented on core-bundle/contao/templates/js/js_accordion.html5:
```diff
@@ -0,0 +1,58 @@
+<?php
+
+// Add the swipe style sheet
```
```suggestion
// Add the handorgel style sheet
```

aschempp committed:
Update core-bundle/contao/templates/js/js_accordion.html5

Co-authored-by: Martin Auswöger <martin@auswoeger.com>

ausi commented on core-bundle/contao/templates/js/js_accordion.html5:
```diff
@@ -0,0 +1,58 @@
+        const button = document.createElement('button');
+        button.type = 'button';
+        button.innerHTML = toggler.innerHTML;
+        toggler.innerHTML = '';
```
```suggestion
        while (toggler.firstChild) {
            button.appendChild(toggler.firstChild);
        }
```
I’d prefer moving the nodes instead of copying HTML strings.

aschempp replied:
sound legit, but couln't we do something like `button.append(toggler.children())` instead?

ausi replied:
Yes, something similar:

```suggestion
        button.append(...toggler.childNodes);
```

leofeyer commented on composer.json:
```diff
@@ -48,6 +48,7 @@
         "contao-components/contao": "^9.1",
         "contao-components/datepicker": "^3.0",
         "contao-components/dropzone": "^5.0.1",
+        "contao-components/handorgel": "^0.5.0",
```
```suggestion
        "contao-components/handorgel": "^0.5",
```

leofeyer commented on core-bundle/composer.json:
```diff
@@ -48,6 +48,7 @@
         "contao-components/contao": "^9.1",
         "contao-components/datepicker": "^3.0",
         "contao-components/dropzone": "^5.0.1",
+        "contao-components/handorgel": "^0.5.0",
```
```suggestion
        "contao-components/handorgel": "^0.5",
```

aschempp committed:
Apply suggestions from code review

Co-authored-by: Leo Feyer <1192057+leofeyer@users.noreply.github.com>

aschempp committed:
Update core-bundle/contao/templates/js/js_accordion.html5

Co-authored-by: Martin Auswöger <martin@auswoeger.com>

leofeyer committed:
Merge branch '5.x' into feature/accordion

leofeyer committed:
CS

leofeyer commented on core-bundle/contao/templates/js/js_accordion.html5:
```diff
@@ -0,0 +1,57 @@
+          return;
+        }
+
+        const button = document.createElement('button');
```
I guess we have to use a button for accessibility, right? Because it sure doesn‘t look nice:

<img width="713" alt="" src="https://github.com/contao/contao/assets/1192057/f7687957-b8f0-4502-9f08-c378c5dac099">

aschempp replied:
yes we do, that's the whole point. But styling a button shouldn't really be a problem, right?

leofeyer commented:
Thank you @aschempp.
